### PR TITLE
Updates to php library dependencies

### PR DESCRIPTION
**Pull request by joelit**
## Reasons for creating this PR

Library dependencies need to be inspected periodically to keep the PHP codebase up-to-date

## Description of the changes in this PR

Changes to composer.json. The changes can be implemented by running `php composer.phar update`. Remaining work can be inspected by running `php composer.phar outdated`. Possible vulnerabilitiy reports can be inspected by running `php composer.phar audit`.

- [x] phpUnit tests pass locally with my changes
- [x] I have added tests that show that the new code works, or tests are not relevant for this PR (e.g. only HTML/CSS changes)
- [x] The PR doesn't reduce accessibility of the front-end code (e.g. tab focus, scaling to different resolutions, use of `.sr-only` class, color contrast)
- [x] The PR doesn't introduce unintended code changes (e.g. empty lines or useless reindentation)

### CODE REVIEW

**Comment by osma** [comment]
PHPUnit now shows these warnings:

```
There were 6 warnings:

1) ConceptPropertyValueLiteralTest::testGetLabelThatIsABrokenDate
Expecting E_WARNING and E_USER_WARNING is deprecated and will no longer be possible in PHPUnit 10.

2) ConceptTest::testGetTimestampInvalidWarning
Expecting E_ERROR and E_USER_ERROR is deprecated and will no longer be possible in PHPUnit 10.

3) VocabularyConfigTest::testGetDefaultLanguageWhenNotSet
Expecting E_ERROR and E_USER_ERROR is deprecated and will no longer be possible in PHPUnit 10.

4) VocabularyConfigTest::testGetDataURLsNotGuessable
Expecting E_WARNING and E_USER_WARNING is deprecated and will no longer be possible in PHPUnit 10.

5) VocabularyConfigTest::testGetDataURLsMarcNotDefined
Expecting E_WARNING and E_USER_WARNING is deprecated and will no longer be possible in PHPUnit 10.

6) VocabularyConfigTest::testGetPropertyOrderUnknown
Expecting E_WARNING and E_USER_WARNING is deprecated and will no longer be possible in PHPUnit 10.
```

This can also be seen in the CI runs, e.g. [this one](https://github.com/NatLibFi/Skosmos/actions/runs/11271889277/job/31345673190).

I think these should be fixed in this PR. It would make future PHPUnit upgrades easier...

File: tests/VocabularyConfigTest.php [hunk 1]
```diff
@@ -17,6 +17,8 @@ protected function setUp(): void
         setlocale(LC_ALL, 'en_GB.utf8');
         $this->model = new Model('/../../tests/testconfig.ttl');
         $this->assertNotNull($this->model->getVocabulary('test')->getConfig()->getPluginRegister(), "The PluginRegister of the model was not initialized!");
+
```
Comment (osma) [inline tests/VocabularyConfigTest.php:20]: Unnecessary whitespace?

Comment (joelit) [reply tests/VocabularyConfigTest.php]: Good call! The trace of me trying to setup a global error handler...
